src/lib/MakeEvalable.ts

```diff
@@ -0,0 +1,167 @@
+const relIndent = " ".repeat(2);
+
+export interface ConstructorFunc<T extends object> {
+  new (...params: unknown[]): T;
+}
+export type FactoryFunc<T extends object> = (...params: unknown[]) => T;
+export type DestructureFunc<T extends object> = (obj: T) => unknown[];
+
+type CustomTypePlugin<T extends object> = {
+  destructureFunc: DestructureFunc<T>;
+  factory?: FactoryFunc<T>;
+  name?: string;
+};
+const customTypePlugins = new Map<
+  ConstructorFunc<object>,
+  CustomTypePlugin<object>
+>();
+
+class AnnotatedToken {
+  constructor(
+    public text: string,
+    public type: "open" | "close" | "delimiter",
+  ) {}
+}
+type Token = string | AnnotatedToken;
+type TokenGenerator = Generator<Token>;
+
+function* makeAttrListTokens(obj: object): TokenGenerator {
+  for (const [attrName, attrValue] of Object.entries(obj!)) {
+    yield `"${attrName.toString()}": `;
+    yield* makeEvalableTokens(attrValue);
+    yield new AnnotatedToken(",", "delimiter");
+  }
+}
+
+function* makeListTokens(obj: unknown[]): TokenGenerator {
+  for (const item of obj) {
+    yield* makeEvalableTokens(item);
+    yield new AnnotatedToken(",", "delimiter");
+  }
+}
+
+function* makeEvalableTokens(obj: unknown): TokenGenerator {
+  if (obj === null) {
+    yield "null";
+  } else if (obj === undefined) {
+    yield "undefined";
+  } else if (typeof obj == "number" && isNaN(obj)) {
+    yield "NaN";
+  } else if (typeof obj == "function") {
+    yield "() => {}"; // has to be replaced by other mechanism
```

Comment:
currently the exception would be ignored and the user wouldn't get any hint whats wrong. 

But in the mid-term this might change (remember we discussed to show excecptions during evaluation to the user)

I will change this...

Reply:
I saw, that events are containing callables => the current workaround is required until a clean solution is available.

Possible clean solution: objects not representable (as i.e. function) are wrapped into i.e. "obj(<number>)". When evaluating them back to js object these objects are converted back by a corresponding plugin...